app/models/hostgroup.rb

```diff
@@ -323,4 +324,14 @@ def validate_subnet_types
     errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if subnet.present? && subnet.type != 'Subnet::Ipv4'
     errors.add(:subnet6, _("must be of type Subnet::Ipv6.")) if subnet6.present? && subnet6.type != 'Subnet::Ipv6'
   end
+
+  def validate_compute_profile
+    return if compute_profile_id.nil?
```

Comment:
this could also be an `if: -> { compute_profile_id.present? }` option on the `validate` deceleration, but just a matter of style, up to you to decide which you prefer.